TYPO3.TypoScript/Classes/TYPO3/TypoScript/Core/Cache/RuntimeContentCache.php

```diff
@@ -302,4 +306,43 @@ public function getEnableContentCache()
     {
         return $this->enableContentCache;
     }
+
+    /**
+     * Cleans incoming content from our cache segment tokens \x02, \x03 and \x1f by replacing valid cache parts temporarily
+     * and then replacing leftover tokens.
+     *
+     * @param string $content
+     * @return string
+     */
+    protected function cleanContentFromExcessCacheMarkers($content)
```

Comment:
Since this code is very specific to the cache segment tokens, I'd suggest to move it to the contentcacheparser as a static method.